File: test/language/expressions/cpp/UsedBeforeTypeTest.cpp
#include "math.h"

#include "gtest/gtest.h"

#include "expressions/used_before_type/UsedBeforeTypeExpression.h"

namespace expressions
{
namespace used_before_type
{

TEST(UsedBeforeTypeTest, bitSizeOfWithOptional)
{
    UsedBeforeTypeExpression usedBeforeTypeExpression;
    usedBeforeTypeExpression.setColor(Color::RED);
    usedBeforeTypeExpression.setIsRedColorLight(true);

    const size_t usedBeforeTypeExpressionBitSizeWithOptional = 8;
    ASSERT_EQ(usedBeforeTypeExpressionBitSizeWithOptional, usedBeforeTypeExpression.bitSizeOf());
}

TEST(UsedBeforeTypeTest, bitSizeOfWithoutOptional)
{
    UsedBeforeTypeExpression usedBeforeTypeExpression;
    usedBeforeTypeExpression.setColor(Color::BLUE);

    const size_t usedBeforeTypeExpressionBitSizeWithoutOptional = 7;
    ASSERT_EQ(usedBeforeTypeExpressionBitSizeWithoutOptional, usedBeforeTypeExpression.bitSizeOf());
}

} // namespace used_before_type
} // namespace expressions

File: test/language/expressions/python/UsedBeforeTypeTest.py
import unittest

from testutils import getZserioApi

class UsedBeforeTypeTest(unittest.TestCase):
    @classmethod
    def setUpClass(cls):
        cls.api = getZserioApi(__file__, "expressions.zs").used_before_type

    def testBitSizeOfWithOptional(self):
        usedBeforeTypeExpression = self.api.UsedBeforeTypeExpression(self.api.Color.RED, True)
        self.assertEqual(self.USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL,
                         usedBeforeTypeExpression.bitSizeOf())

    def testBitSizeOfWithoutOptional(self):
        usedBeforeTypeExpression = self.api.UsedBeforeTypeExpression()
        usedBeforeTypeExpression.color = self.api.Color.BLUE
        self.assertEqual(self.USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL,
                         usedBeforeTypeExpression.bitSizeOf())

    USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITH_OPTIONAL = 8
    USED_BEFORE_TYPE_EXPRESSION_BIT_SIZE_WITHOUT_OPTIONAL = 7
